test/common/http/match_delegate/config_test.cc

```diff
@@ -315,6 +315,63 @@ TEST(DelegatingFilterTest, MatchTreeSkipActionDecodingHeaders) {
   delegating_filter->decodeComplete();
 }
 
+// This function creates the matcher tree only with on_no_match set. So it can not be used with
+// on_match test cases.
+template <class InputType, class ActionType>
+Matcher::MatchTreeSharedPtr<Envoy::Http::HttpMatchingData>
+createMatchTreeWithOnNoMatch(const std::string& name, const std::string& value) {
+  auto tree = std::make_shared<Matcher::ExactMapMatcher<Envoy::Http::HttpMatchingData>>(
+      std::make_unique<InputType>(name),
+      Matcher::OnMatch<Envoy::Http::HttpMatchingData>{
+          []() { return std::make_unique<ActionType>(); }, nullptr});
+
+  // No action is set on match. i.e., nullptr action factory cb.
+  tree->addChild(
+      value, Matcher::OnMatch<Envoy::Http::HttpMatchingData>{[]() { return nullptr; }, nullptr});
+  return tree;
+}
+
+TEST(DelegatingFilterTest, MatchTreeOnNoMatchSkipActionDecodingHeaders) {
+  // The filter is added, but since we match on the request header we skip the filter.
```

Comment:
Maybe elaborate on this explanation a bit more, might also want to put it as a comment on top of the test to explain the scenario that this test is testing